core/src/main/java/io/neow3j/transaction/Transaction.java

```diff
@@ -298,7 +294,6 @@ private void serializeWithoutWitnesses(BinaryWriter writer) throws IOException {
     @Override
     public void serialize(BinaryWriter writer) throws IOException {
         serializeWithoutWitnesses(writer);
-        this.witnesses.sort(Comparator.comparing(Witness::getScriptHash));
```

Comment:
Sorting obviously doesn't work if there are empty witnesses. I also didn't find the sorting anymore in the current neo code.